src/main/java/com/synopsys/integration/detect/lifecycle/boot/DetectBoot.java

```diff
@@ -279,17 +280,29 @@ private BlackduckScanMode decideScanMode(BlackDuckConnectionDetails blackDuckCon
         if(!blackduckScanModeSpecified && autonomousScanEnabled) {
             Optional<String> scaasFilePath = detectConfigurationFactory.getScaaasFilePath();
             Optional<String> blackDuckUrl = blackDuckConnectionDetails.getBlackDuckUrl();
+            BlackduckScanMode blackduckScanMode = detectConfigurationFactory.createScanMode(); // getting scan mode from previous scan
 
             AllEnumList<DetectTool> detectTools = detectConfiguration.getValue(DetectProperties.DETECT_TOOLS);
 
             if (blackDuckUrl.isPresent()) {
                 boolean isNotRapid = detectTools.representedValues().stream().anyMatch(tool -> !rapidTools.contains(tool)) || scanTypeEvidenceMap.keySet().stream().anyMatch(tool -> !rapidTools.contains(tool));
-                if (!scanTypeEvidenceMap.isEmpty() && !isNotRapid && scaasFilePath.isPresent()) {
+                if ((!scanTypeEvidenceMap.isEmpty() && !isNotRapid && scaasFilePath.isPresent()) || blackduckScanMode.displayName().equals("Rapid")) {
```

Comment:
Instead of comparing to a string literal, can we compare to BlackduckScanMode.RAPID? (Same applies to below Stateless comparison).